core/logger/splunk.go

```diff
@@ -0,0 +1,241 @@
+	FORWARDER_DELETED = "FORWARDER_DELETED"
+
+	EXTERNAL_INITIATOR_CREATED = "EXTERNAL_INITIATOR_CREATED"
+	EXTERNAL_INITIATOR_DELETED = "EXTERNAL_INITIATOR_DELETED"
+
+	JOB_PROPOSAL_SPEC_APPROVED = "JOB_PROPOSAL_SPEC_APPROVED"
+	JOB_PROPOSAL_SPEC_UPDATED  = "JOB_PROPOSAL_SPEC_UPDATED"
+	JOB_PROPOSAL_SPEC_CANCELED = "JOB_PROPOSAL_SPEC_CANCELED"
+	JOB_PROPOSAL_SPEC_REJECTED = "JOB_PROPOSAL_SPEC_REJECTED"
+
+	CONFIG_UPDATED              = "CONFIG_UPDATED"
+	CONFIG_SQL_LOGGING_ENABLED  = "CONFIG_SQL_LOGGING_ENABLED"
+	CONFIG_SQL_LOGGING_DISABLED = "CONFIG_SQL_LOGGING_DISABLED"
+	GLOBAL_LOG_LEVEL_SET        = "GLOBAL_LOG_LEVEL_SET"
+
+	JOB_ERROR_DISMISSED = "JOB_ERROR_DISMISSED"
+	JOB_RUN_SET         = "JOB_RUN_SET"
+
+	ENV_NONCRITICAL_ENV_DUMPED = "ENV_NONCRITICAL_ENV_DUMPED"
+
+	UNAUTHED_RUN_RESUMED = "UNAUTHED_RUN_RESUMED"
+)
+
+type splunkLogger struct {
+	logger      Logger
+	splunkToken string
+	splunkURL   string
+	developFlag bool
+	hostname    string
+	localIP     string
+}
+
+func newSplunkLogger(logger Logger, splunkToken string, splunkURL string, hostname string, chainlinkDev bool) Logger {
+	// Initialize and return Splunk logger struct with required state for HEC calls
+	return &splunkLogger{
+		logger:      logger.Named("Splunk"),
+		splunkToken: splunkToken,
+		splunkURL:   splunkURL,
+		developFlag: chainlinkDev,
+		hostname:    hostname,
+		localIP:     getLocalIP(),
+	}
+}
+
+func (l *splunkLogger) Auditf(eventID string, data map[string]interface{}) {
+	// goroutine to async POST to splunk HTTP Event Collector (HEC)
+	go l.postLogToSplunk(eventID, data)
+}
+
+// getLocalIP returns the first non- loopback local IP of the host
+func getLocalIP() string {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return ""
+	}
+	for _, address := range addrs {
+		// check the address type and if it is not a loopback the display it
+		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+			if ipnet.IP.To4() != nil {
+				return ipnet.IP.String()
+			}
+		}
+	}
+	return ""
+}
+
+func (l *splunkLogger) postLogToSplunk(eventID string, data map[string]interface{}) {
+	env := "production"
+	if l.developFlag {
+		env = "develop"
+	}
+
+	// Splunk JSON data
+	splunkLog := map[string]interface{}{
+		"eventID":  eventID,
+		"hostname": l.hostname,
+		"localIP":  l.localIP,
+		"env":      env,
+	}
+	if len(data) != 0 {
+		splunkLog["data"] = data
+	}
+
+	// Wrap serialized audit log map into JSON object `event` for API call
+	serializedArgs, _ := json.Marshal(splunkLog)
+	splunkLog = map[string]interface{}{"event": string(serializedArgs)}
+	serializedSplunkLog, _ := json.Marshal(splunkLog)
+
+	// Send up to HEC log collector
+	httpClient := &http.Client{Timeout: time.Second * 60}
+	req, _ := http.NewRequest("POST", l.splunkURL, bytes.NewReader(serializedSplunkLog))
+	req.Header.Add("Authorization", "Splunk "+l.splunkToken)
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		fmt.Printf("Error sending log to Splunk: %v\n", err)
```

Comment:
Can we log errors back through `l.logger.Error*`?
```suggestion
		l.logger.Errorw("Failed to send audit log to Splunk", "err", err, "splunkLog", splunkLog)
		return
```

Reply:
Good call, now errors are logged to outer logger